### ISSUE TITLE
why

### ISSUE THREAD

**Issue by chuichuiY** [sair-lab/AirSLAM#147]
![2024-09-15 00-11-12屏幕截图](https://github.com/user-attachments/assets/0ceb1662-9d3e-4326-8485-89322fef3160)

**Comment by xukuanHIT** [comment]
@chuichuiY Hi, have you installed OpenCV?

**Comment by chuichuiY** [comment]
> @chuichuiY你好，你安装了OpenCV吗？

yes,I downloaded opencv4.2.0 and this problem occurred when compiling AIRSLAM.

**Comment by xukuanHIT** [comment]
OpenCV is not found when compiling DBoW2. Can you check your installation of the OpenCV? For example, can you find the path of OpenCV ".hpp" and ".so" files on your computer?

**Comment by chuichuiY** [comment]
![image](https://github.com/user-attachments/assets/4c8c6d96-4515-4bfa-8dd5-ed7cd85b2c6b)

**Comment by xukuanHIT** [comment]
They are building results. Have you installed them?

**Comment by xukuanHIT** [comment]
@chuichuiY https://docs.opencv.org/4.x/d7/d9f/tutorial_linux_install.html

**Comment by chuichuiY** [comment]
![image](https://github.com/user-attachments/assets/bc1ecd0c-a3b8-4329-843a-32cc1112daad)


> They are building results. Have you installed them?

**Comment by xukuanHIT** [comment]
Can you find opencv2/opencv.hpp?

**Comment by chuichuiY** [comment]
> Can you find opencv2/opencv.hpp?

yes ,@wp:~/opencv-4.2.0/include/opencv2/opencv.hpp

**Comment by xukuanHIT** [comment]
OK, can you find the OpenCVConfig*.cmake?

**Comment by chuichuiY** [comment]
> OK, can you find the OpenCVConfig*.cmake in /usr/local/cmake/opencv4?

/usr/local/stow/absl/lib/cmake/opencv4/OpenCVConfig-version.cmake
/usr/local/stow/absl/lib/cmake/opencv4/OpenCVConfig.cmake

**Comment by xukuanHIT** [comment]
I am not sure why the Cmakelist can not find your opencv. You can set the path in the Cmakelist using "set(OpenCV_DIR /usr/local/stow/absl/lib/cmake/opencv4)".  You can also create a small project to test whether your opencv can be linked correctly.

**Comment by chuichuiY** [comment]
> I am not sure why the Cmakelist can not find your opencv. You can set the path in the Cmakelist using "set(OpenCV_DIR /usr/local/stow/absl/lib/cmake/opencv4)". You can also create a small project to test whether your opencv can be linked correctly.


Thank you so much. It is true that DBOW2 cannot find my opencv, and it can be compiled successfully after modification. I would like to ask what data sets this can run on, and do I need to use rosbag to run it?

**Comment by chuichuiY** [comment]
![1726675087596](https://github.com/user-attachments/assets/590c8377-8520-4eda-8316-6b26ed2f61c8)

**Comment by xukuanHIT** [comment]
Please use the original dataset. We use  the image name to get the time stamp

**Comment by chuichuiY** [comment]
> ![1726675087596](https://private-user-images.githubusercontent.com/111644261/368645428-590c8377-8520-4eda-8316-6b26ed2f61c8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY3MzAyMzcsIm5iZiI6MTcyNjcyOTkzNywicGF0aCI6Ii8xMTE2NDQyNjEvMzY4NjQ1NDI4LTU5MGM4Mzc3LTg1MjAtNGVkYS04MzE2LTZiMjZlZDJmNjFjOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTE5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkxOVQwNzEyMTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jMTVhMzlhOTZmYzkyNTVkMzJlYjZhYWRlYWJmZmUxMGM2MjI4NjQxNjdmNDlkNmM4NDM0NTI0NmNkYTM3YzdkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.E4ChldczfGFpPvwl85zRKwxl92suC59izhEoPDmQShY)
Is this error due to the data set? I changed the name of the MH01 data and the .launch file as required.

**Comment by xukuanHIT** [comment]
Please use the original dataset. Don't change the image name.
